crazyflie_sim/crazyflie_sim/crazyflie_sil.py

```diff
@@ -21,6 +22,17 @@ def __init__(self, fwstate):
         self.yaw = fwstate.yaw
         self.omega = np.array([fwstate.omega.x, fwstate.omega.y, fwstate.omega.z])
 
+        # compute rotation based on differential flatness
+        thrust = self.acc + np.array([0, 0, 9.81])
+        z_body = thrust / np.linalg.norm(thrust)
+        x_world = np.array([np.cos(self.yaw), np.sin(self.yaw), 0])
+        y_body = np.cross(z_body, x_world)
+        y_body /= np.linalg.norm(y_body)
+        x_body = np.cross(y_body, z_body)
+        x_body /= np.linalg.norm(x_body)
```

Comment:
I don't think this needs to be normalized, does it? it won't change the result eitherways

Reply:
Mathematically it shouldn't. However, there were some numerical issues, which then caused rowan to complain about R not being orthogonal.

Reply:
See the updated code with comments explaining it.

Reply:
I agree with the comments.